### PR TITLE
Update substrate to latest master

### PR DESCRIPTION
**Pull request by geigerzaehler**
Update substrate to latest master

Most of the changes are straightforward renames. Here are some additional details:

* The boilerplate code in `node/src/cli.rs` has been replaced with new code and `node/src/command.rs`
* The storage of the free account balance has been changed. The value is now stored in the `free` field of `AccountData`.
* There is a new `frame_system::Event` variant: `CodeUpdated`. We ignore it when determining the success of a transaction.

### CODE REVIEW

**Comment by NunoAlexandre** [comment]
@geigerzaehler I noticed you force pushed a few times. Let me know when you are 100% done so that I review this.

**Comment by geigerzaehler** [comment]
> @geigerzaehler I noticed you force pushed a few times. Let me know when you are 100% done so that I review this.

It’s done now

**Comment by NunoAlexandre** [comment]
@geigerzaehler, the PR is pretty big. Are the key parts you'd like me to be extra careful going through?

**Comment by geigerzaehler** [comment]
You can ignore the changes to the `Cargo.toml`s and `Cargo.lock` as well as `node/src/cli.rs` and `node/src/command.rs`. The rest of the changes should be straightforward. I also added another commit that eliminates some unnecessary changes and reduces the diff size.

File: runtime/build.rs [hunk 1]
```diff
@@ -11,11 +11,11 @@ fn main() {
...
-        // external memory allocator.
-        "-Clink-arg=--export=__heap_base",
-    );
+
+    WasmBuilder::new()
+        .with_current_project()
+        .with_wasm_builder_from_crates("1.0.9")
+        .export_heap_base()
+        .import_memory()
+        .build()
```
Comment (NunoAlexandre) [inline runtime/build.rs]: sweet

File: runtime/src/registry.rs [hunk 2]
```diff
@@ -47,10 +47,10 @@ pub mod store {
             pub Projects: map hasher(blake2_128_concat) ProjectId => Option<state::Project>;
             // The below map indexes each existing project's id to the
             // checkpoint id that it was registered with.
-            pub InitialCheckpoints: map ProjectId => Option<CheckpointId>;
+            pub InitialCheckpoints: map hasher(blake2_256) ProjectId => Option<CheckpointId>;
```
Comment (NunoAlexandre) [inline runtime/src/registry.rs]: Note to self: Check diff between `blake2_128_concat` and `blake2_256 ` for the given use cases.

File: runtime/src/registry.rs [hunk 3]
```diff
@@ -162,7 +162,7 @@ decl_module! {
             Ok(())
         }
 
-        #[weight = SimpleDispatchInfo::FreeNormal]
+        #[weight = SimpleDispatchInfo::InsecureFreeNormal]
```
Comment (NunoAlexandre) [inline runtime/src/registry.rs]: Is this just a rename or did we need to move to an "Insecure" variant?

Comment (geigerzaehler) [reply runtime/src/registry.rs]: This is just a rename. I guess the intention was to indicate that it is indeed unsafe to use.

Comment (NunoAlexandre) [reply runtime/src/registry.rs]: That's good!

**Pull request merged**